src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -104,6 +97,25 @@ class LedgerImpl(vm: VM, blockchainConfig: BlockchainConfig) extends Ledger with
     blockTxsExecResult
   }
 
+  @tailrec
+  private[ledger] final def executePreparedTransactions(
+    signedTransactions: Seq[SignedTransaction], world: InMemoryWorldStateProxy,
+    blockHeader: BlockHeader, signedTransactionValidator: SignedTransactionValidator,
+    acumGas: BigInt = 0, acumReceipts: Seq[Receipt] = Nil, executed: Seq[SignedTransaction] = Nil)
+  : Either[TxsExecutionError, (BlockResult, Seq[SignedTransaction])] = {
+
+    val result = executeTransactions(signedTransactions, world, blockHeader, signedTransactionValidator, acumGas, acumReceipts)
```

Comment:
Have you considered executing transactions one by one? Do you see a problem with that? The only benefit I can find is that `StateBeforeFailure` wouldn't be necessary so it might be interesting to consider it.